### PR TITLE
refactor: reduce usage of `event.node`

### PR DESCRIPTION
**Pull request by Hebilicious**
### 🔗 Linked issue

Waiting on #22515 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Following up on https://github.com/unjs/h3/pull/438 and the changes in h3 1.8, this PR reduce the usage of `event.node` where possible. 

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/runtime/nitro/renderer.ts [hunk 1]
```diff
@@ -199,38 +199,39 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
   const nitroApp = useNitroApp()
 
   // Whether we're rendering an error page
-  const ssrError = event.node.req.url?.startsWith('/__nuxt_error')
+  const ssrError = event.path?.startsWith('/__nuxt_error')
```
Comment (pi0) [inline packages/nuxt/src/core/runtime/nitro/renderer.ts:202]: I expect path is not optional

File: packages/nuxt/src/core/runtime/nitro/renderer.ts [hunk 2]
```diff
@@ -199,38 +199,39 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
...
     ? getQuery(event) as unknown as Exclude<NuxtPayload['error'], Error>
     : null
 
   if (ssrError && ssrError.statusCode) {
     ssrError.statusCode = parseInt(ssrError.statusCode as any)
   }
 
-  if (ssrError && event.node.req.socket.readyState !== 'readOnly' /* direct request */) {
+  if (ssrError && event.node && event.node.req.socket.readyState !== 'readOnly' /* direct request */) {
```
Comment (pi0) [inline packages/nuxt/src/core/runtime/nitro/renderer.ts:210]: Latest unenv has support `req.__unenv__` flag

Comment (Hebilicious) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:210]: I'm not sure I understand 😓 Do you mean we can just do `if(ssrError)` ?

File: packages/nuxt/src/core/runtime/nitro/error.ts [hunk 3]
```diff
@@ -41,12 +41,11 @@ export default <NitroErrorHandler> async function errorhandler (error: H3Error,
   // JSON response
   if (isJsonRequest(event)) {
     setResponseHeader(event, 'Content-Type', 'application/json')
-    event.node.res.end(JSON.stringify(errorObject))
-    return
+    return errorObject
```
[the remark this answers is not in the document]
Comment (Hebilicious) [reply packages/nuxt/src/core/runtime/nitro/error.ts:44]: Would `return send(event, JSON.stringify(errorObject))` be acceptable ?

[the remark this answers is not in the document]
Comment (pi0) [reply packages/nuxt/src/core/runtime/nitro/error.ts:44]: Sure!

File: packages/nuxt/src/app/composables/ssr.ts [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+import { setResponseStatus as _setResponseStatus, getRequestHeaders } from 'h3'
 import type { NuxtApp } from '../nuxt'
 import { useNuxtApp } from '../nuxt'
 
 export function useRequestHeaders<K extends string = string> (include: K[]): { [key in Lowercase<K>]?: string }
 export function useRequestHeaders (): Readonly<Record<string, string>>
 export function useRequestHeaders (include?: any[]) {
   if (import.meta.client) { return {} }
-  const headers = useNuxtApp().ssrContext?.event.node.req.headers ?? {}
+  const event = useNuxtApp().ssrContext?.event
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/ssr.ts:10]: we can likely tree shake this by doing

```suggestion
  const event = import.meta.server && useNuxtApp().ssrContext?.event
```

File: docs/2.guide/2.directory-structure/1.server.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ Nuxt automatically scans files inside these directories to register API and serv
 
 Each file should export a default function defined with `defineEventHandler()` or `eventHandler()` (alias).
 
-The handler can directly return JSON data, a `Promise` or use `event.node.res.end()` to send a response.
+The handler can directly return JSON data, a `Promise`, a `Response`, or use `event.node.res.end()` to send a response.
```
Comment (pi0) [inline docs/2.guide/2.directory-structure/1.server.md:17]: This shall be added in when we release nitro/h3 stable (~by time of Nuxt 3.7)

File: packages/nuxi/src/commands/analyze.ts [hunk 6]
```diff
@@ -77,7 +77,7 @@ export default defineNuxtCommand({
 
       const serveFile = (filePath: string) => lazyEventHandler(async () => {
         const contents = await fsp.readFile(filePath, 'utf-8')
-        return eventHandler((event) => { event.node.res.end(contents) })
+        return eventHandler(event => send(event, contents))
```
Comment (pi0) [inline packages/nuxi/src/commands/analyze.ts:80]: Later we might use h3 util but good for now

File: packages/nuxt/src/core/runtime/nitro/renderer.ts [hunk 7]
```diff
@@ -199,38 +199,39 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
...
-  let url = ssrError?.url as string || islandContext?.url || event.node.req.url!
+  let url = ssrError?.url as string || islandContext?.url || event.path
 
   // Whether we are rendering payload route
   const isRenderingPayload = PAYLOAD_URL_RE.test(url) && !islandContext
   if (isRenderingPayload) {
     url = url.substring(0, url.lastIndexOf('/')) || '/'
-    event.node.req.url = url
+
+    event._path = url
```
Comment (pi0) [inline packages/nuxt/src/core/runtime/nitro/renderer.ts:234]: This is still onofficial way of overiding event and does not sync with `req.url`. i hope works with vite well

File: packages/vite/src/client.ts [hunk 8]
```diff
@@ -174,7 +174,7 @@ export async function buildClient (ctx: ViteBuildContext) {
 
       await new Promise((resolve, reject) => {
         viteServer.middlewares.handle(event.node.req, event.node.res, (err: Error) => {
-          event.node.req.url = originalURL
+          event._path = originalURL
```
Comment (pi0) [inline packages/vite/src/client.ts:177]: Same for here it is not officially a replacement (i truest you already tested)

**Review by pi0 — APPROVED** [review]
LGTM. Let's try merging with main PR branch (i trust you already tested locally and it works with vite)

**Pull request merged**